src/read.c

```diff
@@ -39,6 +39,8 @@ struct ReaderState {
 
 ScannerState s;
 
+IntrState intr;
```

Comment:
Add an instance of `IntrState` into `ReaderState`; it'll thus be stack-allocated as part of that in `ReadEvalCommand` below